src/queue_pair.rs

```diff
@@ -123,12 +124,45 @@ pub(crate) struct QueuePairCap {
     max_inline_data: u32,
 }
 
+impl QueuePairCap {
+    /// Check the attributes and return error if the preset values exceed the capabilities of
+    /// rdma device.
+    pub(crate) fn check_dev_cap(&self, ctx: &Context) -> io::Result<()> {
```

Comment:
Shall we also check other hardware limitations?

Reply:
Yes, there are lots of attributes can be checked, and I'm not sure how to check some of them.
This check is just for init stage and the most error prone part.
I just add a `TODO` and maybe I should create an issue to track?

Reply:
Yes, please file an issue and then resolve the comment.